addons/hr_expense/hr_expense.py

```diff
@@ -271,10 +271,10 @@ def action_move_create(self, cr, uid, ids, context=None):
             lines = map(lambda x:(0,0,self.line_get_convert(cr, uid, x, exp.employee_id.company_id.partner_id, exp.date_confirm, context=context)), eml)
             journal_id = move_obj.browse(cr, uid, move_id, context).journal_id
             # post the journal entry if 'Skip 'Draft' State for Manual Entries' is checked
-            if journal_id.entry_posted:
-                move_obj.button_validate(cr, uid, [move_id], context)
+            move_obj.button_validate(cr, uid, [move_id], context)
             move_obj.write(cr, uid, [move_id], {'line_id': lines}, context=context)
```

Comment:
i think button_validate() just calls post(), so you can remove the first one because you duplicated the calls here